src/src/rx-serial/SerialHoTT_TLM.cpp

```diff
@@ -0,0 +1,468 @@
+{
+    // indicate external sensor is present
+    telemetry.SetCrsfBatterySensorDetected();
+
+    // prepare CRSF telemetry packet
+    CRSF_MK_FRAME_T(crsf_sensor_battery_t)
+    crsfBatt = {0};
+    crsfBatt.p.voltage = htobe16(getHoTTvoltage());
+    crsfBatt.p.current = htobe16(getHoTTcurrent());
+    crsfBatt.p.capacity = htobe24(getHoTTcapacity() * 10); // HoTT: 1 = 10mAh, CRSF: 1 ? 1 = 1mAh
+    crsfBatt.p.remaining = getHoTTremaining();
+    CRSF::SetHeaderAndCrc((uint8_t *)&crsfBatt, CRSF_FRAMETYPE_BATTERY_SENSOR, CRSF_FRAME_SIZE(sizeof(crsf_sensor_battery_t)), CRSF_ADDRESS_CRSF_TRANSMITTER);
+
+    // send packet only if min rate timer expired or values have changed
+    if ((now - lastBatterySent >= BATT_MIN_CRSFRATE) || (lastBatteryCRC != crsfBatt.crc))
+    {
+        lastBatterySent = now;
+
+        telemetry.AppendTelemetryPackage((uint8_t *)&crsfBatt);
+    }
+
+    lastBatteryCRC = crsfBatt.crc;
+}
+
+// HoTT telemetry data getters
+uint16_t SerialHoTT_TLM::getHoTTvoltage()
+{
+    if (device[EAM].present)
+    {
+        return eam.main_voltage;
+    }
+    else if (device[GAM].present)
+    {
+        return (gam.InputVoltage);
+    }
+    else if (device[ESC].present)
+    {
+        return esc.input_v;
+    }
+
+    return 0;
+}
+
+uint16_t SerialHoTT_TLM::getHoTTcurrent()
+{
+    if (device[EAM].present)
+    {
+        return eam.current;
+    }
+    else if (device[GAM].present)
+    {
+        return gam.Current;
+    }
+    else if (device[ESC].present)
+    {
+        return esc.current;
+    }
+    
+    return 0;
+}
+
+uint32_t SerialHoTT_TLM::getHoTTcapacity()
+{
+    if (device[EAM].present)
+    {
+        return eam.batt_cap;
+    }
+    else if (device[GAM].present)
+    {
+        return (gam.Capacity);
+    }
+    else if (device[ESC].present)
+    {
+        return esc.capacity;
+    }
+
+    return 0;
+}
+
+int16_t SerialHoTT_TLM::getHoTTaltitude()
+{
+    if (device[GPS].present)
+    {
+        return gps.Altitude;
+    }
+    else if (device[VARIO].present)
+    {
+        return vario.altitude;
+    }
+    else if (device[EAM].present)
+    {
+        return eam.altitude;
+    }
+    else if (device[GAM].present)
+    {
+        return gam.Altitude;
+    }
+
+    return 0;
+}
+
+int16_t SerialHoTT_TLM::getHoTTvv()
+{
+    if (device[GPS].present)
+    {
+        return (gps.m_per_sec);
+    }
+    else if (device[VARIO].present)
+    {
+        return (vario.climbrate);
+    }
+    else if (device[EAM].present)
+    {
+        return (eam.climbrate);
+    }
+    else if (device[GAM].present)
+    {
+        return (gam.m_per_sec);
+    }
+
+    return 0;
+}
+
+uint8_t SerialHoTT_TLM::getHoTTremaining()
+{
+    if (device[GAM].present)
+    {
+        return gam.fuel_scale;
+    }
+
+    return 0;
+}
+
+int32_t SerialHoTT_TLM::getHoTTlatitude()
+{
+    if (!device[GPS].present)
+    {
+        return 0;
+    }
+
+    uint8_t deg = gps.Lat_DegMin / 100;
```

Comment:
Extract the 100 to a constant i.e. GPS_UNIT_SCALE or somesuch.

Reply:
added constants to class definition